### PR TITLE
Issues/7272 2fa screen

### PR DESCRIPTION
**Pull request by aerych**
Refs #7272 

This PR apples the remaining visual style to the 2fa view controller and removes its inheritance from the old Signin2FAViewController.

Notes: 
As @hypest pointed in chat, the text in the comp was specific to someone using SMS for their auth code.  I've opted to match what is in the Android app currently rather than what's in the comp.  There might still be room for a slight tweak to the instructional text, so pinging for a copy review as well.
I've also left the status label in place tho I'm not sure we should keep it.  Seems like a hold over from the previous design and not a great fit here.  Thoughts? 

To test:
- Ensure the visual style was correctly implemented.
- Confirm logging in with a 2fa code works correctly.

Needs review: @nheagy 

Examples:

simulator screen shot jun 28 2017 5 22 39 pm
simulator screen shot jun 28 2017 5 23 09 pm
simulator screen shot jun 28 2017 5 29 11 pm

### CODE REVIEW

**Comment by benhuberman** [comment]
@aerych Just wanted to double-check that you're using "authenticator app" generically here, rather than, say, Google Authenticator," in which case you'd need to capitalize "Authenticator."

Otherwise, the only minor tweak I'd suggest is to go with "Text **me** a code instead" as it makes it immediately clear that "text" is an imperative verb here. Alternatively, you could go with "Send me a code via text message [instead]."

**Comment by aerych** [comment]
Thanks @benhuberman! 
> Just wanted to double-check that you're using "authenticator app" generically here

Yep. Used in the generic sense.  I'll make the other tweak you suggested.

**Comment by aerych** [comment]
Okay. I think this one is ready for review now, @nheagy! 
@hypest FYI for the text tweaks we made.

File: WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift [hunk 1]
```diff
@@ -1,40 +1,279 @@
 import UIKit
+import SVProgressHUD
+import WordPressComAnalytics
+import WordPressShared
+
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift]: Not sure why it's missing from this diff, but the class's documentation was removed https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/NUX/Signin2FAViewController.swift#L6

File: WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift [hunk 2]
```diff
@@ -1,40 +1,279 @@
...
+        }
+    }
 
-class Login2FAViewController: Signin2FAViewController, LoginViewController {
-    // let the storyboard's style stay
-    override func setupStyles() {}
 
-    override func dismiss() {
-        self.performSegue(withIdentifier: .showEpilogue, sender: self)
+    class func controller(_ loginFields: LoginFields) -> Login2FAViewController {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift]: This method can be nuked.

Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift]: Oopsie... I had added that while testing the UI (I wired a short cut from the prologue) but forgot to remove it.  I'll give it the axe.

**Comment by folletto** [comment]
> I'm really curious about the error if the wrong code is entered. The old black error is shown, and that thing is so disruptive. I'd prefer the new-style red text. Thoughts?

+1
If there's not technical limitation or reason for that, it would be ideal.

> This screen may also be worth having a new-style 'help' dialog 

+1
If you pop me a current screenshot for the old "help" I can check for the new one. :)

**Comment by aerych** [comment]
> I'd prefer the new-style red text. Thoughts?

Agreed.  That's analogous to an invalid password and we should be using the red error label for that according to the error doc. I'll make this tweak.

**Comment by nheagy** [comment]
@folletto there wasn't any help specific to this screen in the old flow, fwiw.

**Comment by aerych** [comment]
@nheagy ready for another round sir!

**Comment by nheagy** [comment]
@aerych looks great!

:shipit: 

…I'm going to have fun merging this with `#7497` 😜

**Comment by folletto** [comment]
> there wasn't any help specific to this screen in the old flow, fwiw.

Ok let's proceed without then! 🚢

**Comment by aerych** [comment]
Thanks all!

**Pull request merged**